src/main/java/org/sonar/plugins/github/GlobalReport.java

```diff
@@ -97,20 +101,19 @@ public String getStatusDescription() {
     return sb.toString();
   }
 
-  public GHCommitState getStatus() {
-    return (newIssues(Severity.BLOCKER) > 0 || newIssues(Severity.CRITICAL) > 0) ? GHCommitState.ERROR : GHCommitState.SUCCESS;
+  public GHCommitState getStatus(CommitStatusResolver commitStatusResolver) {
+    return commitStatusResolver.getStatus(newIssuesBySeverity);
   }
 
   private int newIssues(Severity s) {
     return newIssuesBySeverity[s.ordinal()];
   }
 
   private void appendSummaryBySeverity(ReportBuilder builder) {
-    appendNewIssues(builder, Severity.BLOCKER);
-    appendNewIssues(builder, Severity.CRITICAL);
-    appendNewIssues(builder, Severity.MAJOR);
-    appendNewIssues(builder, Severity.MINOR);
-    appendNewIssues(builder, Severity.INFO);
+    int from = 0;
+    int to = Severity.values().length - 1;
+    IntStream.rangeClosed(from, to).forEach(j -> printNewIssuesForMarkdown(sb, Severity.values()[to - j]));
```

Comment:
I don't see the `sb` object in this function, could you please fix that? + In general, this functional looks interesting for me, as currently we using this github-plugin with a custom patch to break the build on major issues

Reply:
ready!